### PR TITLE
Change WindowBuilder to a more conventional builder pattern

### PR DESCRIPTION
**Pull request by azymohliad**
Following up on [the discussion in Zulip](https://xi.zulipchat.com/#narrow/stream/351333-glazier/topic/Builder.20pattern.20in.20WindowBuilder), this PR changes `WindowBuilder` methods `&mut self` argument to `mut self` and return value to `Self`, so that methods could be chained. Besides that, I took a liberty to also remove `set_` prefix from `WindowBuilder` methods, and also changed `WindowBuilder` for all backends to keep it consistent with the top-level one. Let me know if any of that is undesired, I'll revert.

I've tested all examples except `accesskit` on X11 backend, and tested their compilation on Wayland and GTK backends (I have issues running on these backends same as on `main`: GTK segfaults, Wayland has transparent window). I hope the CI will make sure others compile?

### CODE REVIEW

**Comment by jaredoconnell** [comment]
One thing I've noticed in the other code in the projects here is often both the builder pattern functions and the basic `set_` functions have been available.
Because the functions are now returning `Self`, it moves the value, so if you don't want to use the builder pattern like this, you still need to re-assign the values, which is annoying and could confuse beginners. That's why it may make sense to have them both.

If you have them both, you can still reduce redundancy by calling the set_ functions from within the builder pattern functions, reducing the possibility of accidental differences in behavior between the two functions.

What do others think?

**Comment by azymohliad** [comment]
> so if you don't want to use the builder pattern like this, you still need to re-assign the values

Hm.. I didn't think of that. What are some use cases where one wouldn't want to use builder methods moving the value? Wrapping into another builder is an example I can think of (like the top-level `WindowBuilder` wraps the backend's one), but I'd guess it's fairly uncommon in the application code. 

I guess some extra cost of providing both ways is, if we want to keep all backends `WindowBuilder`-s consistent with the top-level one then we'd have to provide both ways for all of them. But maybe they don't need to have exactly the same API, idk.

**Review by lord — APPROVED** [review]
This looks good! I agree that there's an extra cost for us provide both ways. I think both the naming here and the `mut self -> Self`-style builder matches what [multiple](https://github.com/linebender/piet/blob/5c505b083675c686c2cee5ffc4c4a1cd24fedf9b/piet-svg/src/text.rs#L125-L169) other [builders](https://github.com/linebender/vello/blob/4c5ff6b5bf1bf6b144c0c20ea2c87e7c4f2595ce/src/scene.rs#L205-L310) do in the linebender extended universe, so maybe we can just merge this and see if the lack of `set_` methods becomes difficult for ppl in practice?

I'm not certain who actually has glazier merge permissions, so feel free to comment if you don't and I'd be happy to press the button

**Comment by azymohliad** [comment]
Thanks for a quick review. I think I can merge. Should I wait for more reviews? Also, when ready, should I just merge or squash & merge?

**Comment by jneem** [comment]
One approval is enough, so go ahead and merge when you're ready. Usually we squash, but if you have artisanal hand-crafted commits that you want to preserve, a rebase+merge is also ok.

**Pull request merged**
